queries.sql

```diff
@@ -4,96 +4,119 @@
+        lastName=$(),                       --UPDATE
+        emailAddress=$(),                   --UPDATE
+        phone=$(),                          --UPDATE
+        username=$(username),
+        password=$(password),
+        dob=$(),                            --UPDATE
+        profileURL=$(),                     --UPDATE
+        hobby=$(),                          --UPDATE
+        habitGoal=$(),                      --UPDATE
+        notifications=$(notifications),
+        theme=$(theme)
+
+    WHERE ID = ${req.params.ID}
 
 --Empty Habits 
 ---NEED TO READ: None
 ---NEED TO WRITE: habits, written specifically to the userprofile
 
-SELECT 
-    FROM
+UPDATE UserTable
+    SET firstName=$(firstName),
+        lastName=$(lastName),
+        emailAddress=$(emailAddress),
+        phone=$(phone),
+        username=$(username),
+        password=$(password),
+        dob=$(dob),
+        profileURL=$(profileURL),
+        hobby=$(hobby),
+        habitGoal=$(),                      --UPDATE
+        notifications=$(notifications),
+        theme=$(theme)
+
+    WHERE ID = ${req.params.ID}
 
-    ORDER BY 
 
 --Empty Profile
 ---NEED TO READ: None
 ---NEED TO WRITE: Writing all user profile data. Similar to the edit profile. Current user data, firstName, lastName, habitGoal, habitCategory, hobby, email, profile url
 
-SELECT 
-    FROM
-
-    ORDER BY 
-
---Habit Selector
----NEED TO READ: user data: HabitCategory
----NEED TO WRITE: None
-SELECT 
-    FROM 
-
-    ORDER BY 
+UPDATE UserTable
+    SET firstName=$(),                      --UPDATE
+        lastName=$(),                       --UPDATE
+        emailAddress=$(),                   --UPDATE
+        phone=$(),                          --UPDATE
+        username=$(username),
+        password=$(password),
+        dob=$(),                            --UPDATE
+        profileURL=$(),                     --UPDATE
+        hobby=$(),                          --UPDATE
+        habitGoal=$(),                      --UPDATE
+        notifications=$(notifications),
+        theme=$(theme)
+
+    WHERE ID = ${req.params.ID}
 
 --Habit Tracker
 ---NEED TO READ: N/A
 ---NEED TO WRITE: N/A
 SELECT *
-    FROM Habit, UserHabit, HabitCategory, Username
-    WHERE userID = Username.ID
-        AND habitID = Habit.ID
+    FROM 
+    WHERE 
 
 --Home
 ---NEED TO READ: List of buddies, days of habits tracked, user's habit
 ---NEED TO WRITE: habit stacker information
-SELECT habit, username, buddy1, buddy2
-    FROM UserBuddy, Username, Habit, HabitCategory, UserHabit
+SELECT habit, firstName, lastName, buddy1, buddy2
+    FROM UserTable, Habit, Buddies
     WHERE userID = Username.ID
         AND habitID = Habit.ID
         AND buddy1 != Username.ID
         AND buddy2 != Username.ID
 
 --Login
 ---NEED TO READ: user data: username, password
----NEED TO WRITE: None
-SELECT 
-    FROM
-    ORDER BY 
+---NEED TO WRITE: None, maybe create user upon sign up 
```

Comment:
I'm not sure this has anything to do with the schema, but I was thinking of also implementing a "Forgot Password" kinda thing, but it's just an extra idea. I guess we would have to implement some kind of email verification thing first...ooo new research project!